src/lib/blockchains/namecoin.coffee

```diff
@@ -24,30 +25,74 @@ module.exports = (dnschain) ->
     RCODE_NAME = dns2.consts.RCODE_TO_NAME
     BLOCKS2SEC = 10 * 60
 
-    LOCALHOSTS = _.uniq [
-        "127.0.0.", "10.0.0.", "192.168.", "::1", "fe80::"
-        gConf.get('dns:host'), gConf.get('http:host')
-        gConf.get('dns:externalIP'), gExternalIP()
-        gConf.nmc.get('rpcconnect')
-    ].filter (o)-> typeof(o) is 'string'
-
-    # instanceNum = 0 # debugging
-
-    # It is the hander's job to add answers to 'res' but *NOT* to send them!
-    # Instead, it should call the callback function 'cb'.
-    # Pass in a NAME_RCODE to 'cb' on error, or nothing on success.
-    # 
-    # IMPORTANT: these functions __*MUST*__ be bound to the DNSServer instance 
-    #            that calls them!
-    dnsTypeHandlers =
-        namecoin:
+    # Specifications listed here:
+    # - https://wiki.namecoin.info/index.php?title=Welcome
+    # - https://wiki.namecoin.info/index.php?title=Domain_Name_Specification#Importing_and_delegation
+    # - https://wiki.namecoin.info/index.php?title=Category:NEP
+    # - https://wiki.namecoin.info/index.php?title=Namecoin_Specification
+    VALID_NMC_DOMAINS = /^[a-zA-Z]+\/.+/
+
+    class NamecoinResolver extends BlockchainResolver
+        constructor: (@dnschain) ->
+            @log = gNewLogger 'NMC'
+            @name = 'namecoin'
+            @tld = 'bit'
+
+        config: ->
+            @log.debug "Loading #{@name} resolver"
+
+            gConf.add @name, _.map(_.filter([
+                [process.env.APPDATA, 'Namecoin', 'namecoin.conf']
+                [process.env.HOME, '.namecoin', 'namecoin.conf']
+                [process.env.HOME, 'Library', 'Application Support', 'Namecoin', 'namecoin.conf']
+                ['/etc/namecoin/namecoin.conf']]
+            , (x) -> !!x[0])
+            , (x) -> path.join x...)
+
+            # we want them in this exact order:
+            params = ["port", "connect", "user", "password"].map (x) => gConf[@name].get 'rpc'+x
+            if not _.every params
+                @log.info "#{@name} disabled. (namecoin.conf not found, or rpcuser, rpcpassword, rpcconnect, rpcport not found)"
+                return
+            gConf[@name].set 'host', gConf[@name].get('rpcconnect')
+            @peer = rpc.Client.$create(params...) or gErr "rpc create"
+
+            # TODO: $create doesn't actually connect. you need to open a raw socket
+            #       or an http socket and see if that works before declaring it works
+            @log.info "rpc to namecoind on: %s:%d", params[1], params[0]
+            @
+
+        shutdown: ->
+            @log.debug 'shutting down!'
+            # @peer.end() # TODO: fix this!
+
+        resolve: (path, options, cb) ->
+            result = @resultTemplate()
+            if S(path).endsWith(".#{@tld}") # naimcoinize Domain
```

Comment:
`path` is either xxxx.bit or xxxx
if its called as a DNS request (with the .bit), then we append d/ to `path`,
the else is to do nothing, as we assume the namespace has been properly set (d/, id/, etc) and is already in `path`.